src/codecs/webp/extended.rs

```diff
@@ -0,0 +1,683 @@
+                    let blend_alpha = blend_alpha_f64 as u8;
+
+                    let blend_rgb: [u8; 3] = if blend_alpha == 0 {
+                        [0, 0, 0]
+                    } else {
+                        let mut rgb = [0; 3];
+                        for i in 0..3 {
+                            let canvas_f64 = f64::from(canvas[i]);
+                            let buffer_f64 = f64::from(buffer[index + i]);
+
+                            let val = (buffer_f64 * buffer_alpha
+                                + canvas_f64 * canvas_alpha * (1.0 - buffer_alpha / 255.0))
+                                / blend_alpha_f64;
+                            rgb[i] = val as u8;
+                        }
+
+                        rgb
+                    };
+
+                    Rgba([blend_rgb[0], blend_rgb[1], blend_rgb[2], blend_alpha])
+                } else {
+                    Rgba([
+                        buffer[index],
+                        buffer[index + 1],
+                        buffer[index + 2],
+                        buffer[index + 3],
+                    ])
+                };
+            }
+        }
+
+        let delay = Delay::from_numer_denom_ms(anim_image.duration, 1);
+        let img = canvas.clone();
+        let frame = Frame::from_parts(img, 0, 0, delay);
+
+        if anim_image.dispose {
+            for x in 0..anim_image.width {
+                for y in 0..anim_image.height {
+                    let canvas_index = (x + anim_image.offset_x * 2, y + anim_image.offset_y * 2);
+                    canvas[canvas_index] = background_color;
+                }
+            }
+        }
+
+        Some(Ok(frame))
+    }
+
+    pub(crate) fn fill_buf(&self, buf: &mut [u8]) {
+        match &self.image {
+            ExtendedImageData::Animation { frames, .. } => {
+                //will always have at least one frame
+                frames[0].image.fill_buf(buf);
+            }
+            ExtendedImageData::Static(image) => {
+                image.fill_buf(buf);
+            }
+        }
+    }
+
+    pub(crate) fn get_buf_size(&self) -> usize {
+        match &self.image {
+            ExtendedImageData::Animation { frames, .. } => {
+                //will always have at least one frame
+                frames[0].image.get_buf_size()
+            }
+            ExtendedImageData::Static(image) => image.get_buf_size(),
+        }
+    }
+}
+
+#[derive(Debug)]
+enum WebPStatic {
+    Lossy(RgbaImage),
+    Lossless(LosslessFrame),
+}
+
+impl WebPStatic {
+    pub(crate) fn from_alpha_lossy(
+        alpha: AlphaChunk,
+        vp8_frame: VP8Frame,
+    ) -> ImageResult<WebPStatic> {
+        if alpha.data.len() != usize::from(vp8_frame.width * vp8_frame.height) {
+            return Err(DecoderError::SizeMismatch.into());
+        }
+
+        let mut image_vec =
+            vec![0u8; usize::from(vp8_frame.width) * usize::from(vp8_frame.height) * 4];
```

Comment:
For 32-bit targets this is prone to an overflow with `width = height = u16::MAX` or similar. Prefer checked multiplication and maybe perform that length calculation before the previous statement, as success certifies that the previous is free of overflow as well.